src/python/zquantum/core/wip/circuit/_circuit_test.py

```diff
@@ -0,0 +1,743 @@
+                ]
+            ),
+        ],
+        [
+            Circuit(
+                gates=[
+                    CustomGate(
+                        matrix=sympy.Matrix(
+                            [
+                                [sympy.Symbol("theta_0"), 0],
+                                [0, sympy.Symbol("theta_1")],
+                            ]
+                        ),
+                        qubits=(0,),
+                    )
+                ]
+            ),
+            Circuit(
+                gates=[
+                    CustomGate(
+                        matrix=sympy.Matrix(
+                            [
+                                [sympy.Symbol("gamma_0"), 0],
+                                [0, sympy.Symbol("gamma_1")],
+                            ]
+                        ),
+                        qubits=(0,),
+                    )
+                ]
+            ),
+        ],
+    ],
+)
+def test_gate_eq_not_same_gates(circuit1, circuit2):
+    """The Circuit class should be able to be able to compare two unequal circuits"""
+    # When
+    are_equal = circuit1 == circuit2
+
+    # Then
+    assert not are_equal
+
+
+#### __add__ ####
+@pytest.mark.parametrize(
+    "circuit1, circuit2, expected_circuit",
+    [
+        [
+            Circuit(gates=[]),
+            Circuit(gates=[H(0)]),
+            Circuit(gates=[H(0)]),
+        ],
+        [
+            Circuit(gates=[]),
+            Circuit(gates=[]),
+            Circuit(gates=[]),
+        ],
+        [
+            Circuit(gates=[H(0)]),
+            Circuit(gates=[]),
+            Circuit(gates=[H(0)]),
+        ],
+        [
+            Circuit(gates=[H(0), CNOT(0, 1)]),
+            Circuit(gates=[CNOT(0, 1), H(0)]),
+            Circuit(gates=[H(0), CNOT(0, 1), CNOT(0, 1), H(0)]),
+        ],
+        [
+            Circuit(gates=[H(0), CNOT(0, 1), CustomParameterizedGate]),
+            Circuit(gates=[CNOT(0, 1), H(0)]),
+            Circuit(
+                gates=[
+                    H(0),
+                    CNOT(0, 1),
+                    CustomParameterizedGate,
+                    CNOT(0, 1),
+                    H(0),
+                ]
+            ),
+        ],
+    ],
+)
+def test_add_circuits(circuit1, circuit2, expected_circuit):
+    """The Circuit class should be able to handling adding circuits together"""
+    # When
+    new_circuit = circuit1 + circuit2
+
+    # Then
+    assert new_circuit == expected_circuit
+
+
+#### evaluate ####
+def test_circuit_evaluate_with_all_params_specified():
+    # Given
+    symbols_map = {"theta_0": 0.5, "theta_1": 0.6}
+    RYGateQubit0 = RY(0).evaluate(symbols_map)
+    RZGateQubit0 = RZ(0).evaluate(symbols_map)
+    RZGateQubit0DifferentAngle = RZ(0).evaluate({"theta_1": 0.4})
+    circuit = Circuit(
+        gates=[
+            RX(0),
+            RYGateQubit0,
+            RZGateQubit0,
+            RZGateQubit0DifferentAngle,
+        ]
+    )
```

Comment:
Actually, I think this is incorrect. The gates listed here should be `RX(0), RY(0), RZ(0), RZGateQubit0DifferentAngle`. Otherwise, we aren't really "evaluating" any of the gates when we call `circuit.evaluate(symbols_map)` since all parameterized gates are already evaluated

Reply:
Or perhaps that is the point of this test? If so, there should be another one where this case is addressed.

Reply:
Reimplemented the test case with @dexter2206 in 4f597d3